typescript/cli/src/config/chain.ts

```diff
@@ -69,6 +70,30 @@ export async function createChainConfig({
     default: name[0].toUpperCase() + name.slice(1),
   });
 
+  const technicalStack = (await select({
+    choices: Object.entries(ChainTechnicalStack).map(([_, value]) => ({
+      value,
+    })),
+    message: 'Select the chain technical stack',
+    pageSize: 10,
+  })) as ChainTechnicalStack;
+
+  const arbitrumNitroMetadata: Partial<ChainMetadata> = {};
+  if (technicalStack === ChainTechnicalStack.ArbitrumNitro) {
+    const indexFrom = await detectAndConfirmOrPrompt(
+      async () => {
+        return (await provider.getBlockNumber()).toString();
+      },
+      `Enter`,
+      'starting block number for indexing',
+      'JSON RPC provider',
+    );
+
```

Comment:
@paulbalaji Here is the update, I tested this on curtis network, and it gets the correct L2 block number.

This is what the prompt looks like for block number:

```
? Detected starting block number for indexing as 14053484 from JSON RPC provider, is this correct? no
? Enter starting block number for indexing: 14053484
```

Reply:
@paulbalaji  To resolve [this comment](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/5049#discussion_r1892952213)

I moved chainTechStack prompt after testnet & 'Picked' index from ChainMetadata for Arbitrum Nitro.